pkg/reconciler/pipeline/dag/dag_test.go

```diff
@@ -549,6 +550,78 @@ func TestBuild_InvalidDAG(t *testing.T) {
 	}
 }
 
+func TestBuildGraphWithHundredsOfTasks_Success(t *testing.T) {
```

Comment:
Maybe some performance check too?

Reply:
yes, please so that we do not run into it in future, thanks!

Reply:
The problem with performance checks is that they could be inherently flaky because of the variable performance of the test nodes. If we do add performance checks, I would suggest for now to only log the execution time. 
We can collect such timings over time for some time and then decide on an acceptable bar for the execution time.

Reply:
That's a great idea @afrittoli, where do we store those timings? 

I was running the test you had introduced https://github.com/tektoncd/pipeline/pull/3524 (a huge thanks to you for introducing such test):

https://github.com/tektoncd/pipeline/blob/99404d532ae9e9149cd941d11b754035cfb646c9/pkg/reconciler/pipelinerun/resources/pipelinerunstate_test.go#L1404

I ran it locally yesterday multiple times which took 60 seconds 😲 (without this PR). 

At the time introducing this test, it took less than the default timeout 30 seconds based on the PR description (if I am reading it right):

```This changes adds a unit test that reproduces the issue in https://github.com/tektoncd/pipeline/issues/3521, which
used to fail (with timeout 30s) and now succeedes for pipelines roughly
up to 120 tasks / 120 links. On my laptop, going beyond 120 tasks/links
takes longer than 30s, so I left the unit test at 80 to avoid
introducing a flaky test in CI. There is still work to do to improve
this further, some profiling / tracing work might help.```

There has been many changes introduced since then, we really need a way to flag us (nightly performance tests) when we introduce any delay.

Reply:
Nightly performance test or like you are suggesting to collect timings over time. I am fine with either option.

Reply:
Yes. The `TestBuildGraphWithHundredsOfTasks_Success` in a way is a kind of performance test (like the one I added at the time), because if things slow down significantly, the tests will eventually timeout.
If we collected test execution times and graphed them over time - or if we had some dedicated nightly performance test, we would be able to see a change in execution time sooner than just waiting for the tests to timeout.

That is something that we would need to setup as part of the CI infrastructure. Would you like to create an issue about that?

Reply:
yup definitely, we had a PR from @guillaumerose https://github.com/tektoncd/pipeline/pull/4378 which didn't materialize but at least most of us including @vdemeester and @imjasonh were on board with the idea of running nightly performance tests.